### ISSUE TITLE
Add brew and apt support

### ISSUE THREAD

**Issue by dbalatero** [samphippen/rubyfmt#349]
Hi @reese, hope you're good! Wondering if there are plans to add this as a Cargo package so you can `cargo install rubyfmt`?

I have a few other linting plugins that work that way that I use from Neovim ([example](https://github.com/dbalatero/dotfiles/blob/22b990a44c6998d920da33ccde899b8ba65bf6d6/installers/neovim#L88-L98)), and they seem to nicely make their way to my `$PATH` without much more work. Thanks!

**Comment by reese** [comment]
I feel like we should have a better distribution strategy, but I don't think any plans are set at the moment. (cc @penelopezone if you've got thoughts on this.)

My first thought is that having to actually build the Ruby checkout might make something like `cargo install` kinda annoying unless we got rid of all of the build-time system dependencies (like `make` and `autoreconf`). It might end up being more straightforward to have precompiled binaries and to use system package managers like `brew` instead, but I definitely think having _some_ kind of easier package management is something we should do.

**Comment by dbalatero** [comment]
I think `brew install rubyfmt` / `apt-get install rubyfmt` would be totally fine ways to distribute it. I'm thinking of integrating it into my company's pipeline, and I really just need a cheap, no-nonsense way to:

* get it into CI pipelines
* get it onto dev machines

Honestly, the package manager approach would probably be most familiar to folks, and `cargo` would be a bit more niche unless you do Rust!

**Comment by fables-tales** [comment]
Strong 👍 we should do brew and apt but not cargo

**Comment by matthewford** [comment]
brew fails if you run `make release`

    bash -c "find target/release | grep -i 'librubyfmt-.*\.a' | xargs rm; exit 0"
    find: target/release: No such file or directory

**Comment by reese** [comment]
@matthewford how is brew involved here? What's the original command you're running?

**Comment by matthewford** [comment]
@reese my recipe is here: https://github.com/matthewford/homebrew-core/tree/bump-rubyfmt-0.8

I don't think it's a brew issue specifically I see there was an old closed issue with the same error

**Comment by matthewford** [comment]
https://github.com/penelopezone/rubyfmt/issues/340

**Comment by matthewford** [comment]
https://github.com/Homebrew/homebrew-core/pull/88103#issuecomment-954591471 this is the same issue

**Comment by matthewford** [comment]
@reese using your homebew PR I see a similar issue:
```
Last 15 lines from /Users/matthewford/Library/Logs/Homebrew/rubyfmt/01.cargo:
   Compiling rubyfmt v0.8.0-pre (/private/tmp/rubyfmt-20221120-87784-1mvjnl7/rubyfmt-0.8.0/librubyfmt)
   Compiling unicode-normalization v0.1.21
   Compiling addr2line v0.17.0
   Compiling aho-corasick v0.7.18
   Compiling object v0.29.0
   Compiling bstr v0.2.17
error: failed to run custom build command for `rubyfmt v0.8.0-pre (/private/tmp/rubyfmt-20221120-87784-1mvjnl7/rubyfmt-0.8.0/librubyfmt)`

Caused by:
  process didn't exit successfully: `/private/tmp/rubyfmt-20221120-87784-1mvjnl7/rubyfmt-0.8.0/target/release/build/rubyfmt-072919fc54932cda/build-script-build` (exit status: 1)
  --- stderr
  fatal: not a git repository (or any of the parent directories): .git
  Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
warning: build failed, waiting for other jobs to finish...
error: failed to compile `rubyfmt-main v0.8.0-pre (/private/tmp/rubyfmt-20221120-87784-1mvjnl7/rubyfmt-0.8.0)`, intermediate artifacts can be found at `/private/tmp/rubyfmt-20221120-87784-1mvjnl7/rubyfmt-0.8.0/target`
```

**Comment by reese** [comment]
I think you're pointing to several different issues here, the issue on my PR, #340, and the `make` error you're referring to are all separate issues.

**Comment by reese** [comment]
Mine is because we rely on a git checkout, so we probably need to actually pull the submodule first, your error is because `make release` calls `find` which exits non-zero the first time you build

**Comment by reese** [comment]
#340 is specific to M1 and is referring to an error in the build script, that's separate from either of these

**Comment by matthewford** [comment]
@reese I got around the it's not a git dir, by cloning in this commit https://github.com/matthewford/homebrew-core/commit/649219cf1e43edb432d4f0b1dee442e41a18350c, but then ran into the other issue about `find: target/release: No such file or directory`. Not on an M1 mac

**Comment by reese** [comment]
Right, I'm saying the M1 bit is not what you're running into. You can either just `mkdir_p "target/release"` or just use `cargo install` like on my PR instead. (Are you committing to doing the brew bit here? I'm happy to leave this to you if you want, we don't need to both be working on this)

**Comment by matthewford** [comment]
@reese I'm struggling to get this to compile via homebrew, I tried adding 

    mkdir_p "target/release"

but then still getting this error:

  Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }

**Comment by reese** [comment]
Can you give the full error output? There should be more above that line.

**Comment by matthewford** [comment]
Unfortunately there isn't any 

 ```
  Compiling rubyfmt v0.8.0-pre (/private/tmp/rubyfmt-20221120-20601-1tb15x5/librubyfmt)
   Compiling unicode-normalization v0.1.21
   Compiling aho-corasick v0.7.18
error: failed to run custom build command for `rubyfmt v0.8.0-pre (/private/tmp/rubyfmt-20221120-20601-1tb15x5/librubyfmt)`

Caused by:
  process didn't exit successfully: `/private/tmp/rubyfmt-20221120-20601-1tb15x5/target/release/build/rubyfmt-072919fc54932cda/build-script-build` (exit status: 1)
  --- stderr
  Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
warning: build failed, waiting for other jobs to finish...
make: *** [release] Error 101
```

**Comment by reese** [comment]
I don't think I have any immediate answers then, I'm debugging the same thing but it builds fine outside of brew, so it's definitely something related to the brew setup (which I'm not at all familiar with).

**Comment by matthewford** [comment]
If you use the debug flag to keep the tmp dir around.

`brew reinstall --build-from-source rubyfmt -v --debug
`
`make release` works fine within the tmp dir, so I have no idea

**Comment by matthewford** [comment]
@reese that last commit works well. Thanks for this!

**Comment by reese** [comment]
Looks like you already saw it, but just for posterity in case anyone else sees this, I'm updating the brew formula [here](https://github.com/Homebrew/homebrew-core/pull/116203), and I _think_ the missing files mentioned in the thread were probably `autoconf` and `make` not being marked as dependencies, so I think just adding them should get us working.

@matthewford thanks for taking a look! I definitely just took all the git bits from your branch to make the submodule stuff work, your help was much appreciated!

**Comment by reese** [comment]
Ah dang, so one issue here:

@penelopezone I don't think `brew` will let us release a pre-release version. We'd probably need to do a `0.8.0` release that's not `0.8.0-pre` if we want it in brew.

**Comment by fables-tales** [comment]
I mean call it 0.8.1 and ship it I guess? That seems like a very weird arbitrary constraint but I have no issues with that

**Comment by reese** [comment]
We might just be able to un-check the `prerelease` option on the Github release, I don't know if it actually requires us to cut a new release.

**Comment by reese** [comment]
I'm just gonna try unchecking it and see if that works 🤷

**Comment by fables-tales** [comment]
why is homebrew like this

**Comment by reese** [comment]
Ok yeah just unchecking the box worked, thanks brew very cool. Now just gotta make sure all the builds work

**Comment by reese** [comment]
Just an update for the folks following along at home, the current release is now out on `brew`, so now you can `brew install rubyfmt`!

Still a WIP on getting `apt` support working though, that's coming down the line.

**Comment by dbalatero** [comment]
For anyone else installing, I had to run `brew update` to pick up the 0.8.0 version (I guess there's an older 0.2.0 version that was on brew before)!
